bottomless/src/replicator.rs

```diff
@@ -1402,38 +1388,33 @@ impl Replicator {
                     }
                 }
                 let frame = self.get_object(key.into()).send().await?;
-                let mut frameno = first_frame_no;
                 let mut reader = BatchReader::new(
-                    frameno,
+                    first_frame_no,
                     tokio_util::io::StreamReader::new(frame.body),
                     self.page_size,
                     compression_kind,
                 );
 
                 while let Some(frame) = reader.next_frame_header().await? {
-                    let pgno = frame.pgno();
-                    let page_size = self.page_size;
+                    last_received_frame_no = reader.next_frame_no();
                     reader.next_page(&mut page_buf).await?;
                     if self.verify_crc {
                         checksum = frame.verify(checksum, &page_buf)?;
                     }
-                    pending_pages.insert(pgno, &page_buf).await?;
-                    if frame.is_committed() {
-                        let pending_pages = std::mem::replace(
-                            &mut pending_pages,
-                            TransactionPageCache::new(
-                                self.restore_transaction_page_swap_after,
-                                page_size as u32,
-                                self.restore_transaction_cache_fpath.clone(),
-                            ),
-                        );
-                        pending_pages.flush(db).await?;
-                        applied_wal_frame = true;
-                    }
-                    frameno += 1;
-                    last_received_frame_no += 1;
+                    let (crc1, crc2) = frame.crc();
+                    let checksum = (crc1 as u64) << 32 | crc2 as u64;
+                    let frame_to_inject = libsql_replication::frame::Frame::from_parts(
+                        &libsql_replication::frame::FrameHeader {
```

Comment:
Technically we have another `bottomless::was::WalFrameHeader` definition used by bottomless. We could off get rid of it and use only one.

Reply:
(oh and that code landed in #915 by mistake :sweat_smile: it belongs to #928, moved)